// react-leetcode/dist/components/recentSubmissions/recentSubmission.js
import { j as e } from "../../index-CuHmWrv0.js";
import { forwardRef as m, useCallback as f, memo as u } from "react";
import x from "../../utils/leetcodeQuery.js";
import { useFetch as p } from "../../hooks/index.js";
import b from "../LoadingOrError.js";
const j = m(({
  userName: l,
  loadingComponent: c,
  theme: s = {
    primaryColor: "rgba(34,211,238,1)",
    secondaryColor: "rgba(209,213,219,1)",
    bgColor: "rgba(68,64,60,1)"
  }
}, a) => {
  const n = f(() => x.fetchUserRecentSubmissions(l), [l]), { data: o, loading: t, error: i } = p(n);
  return t || i || !o ? /* @__PURE__ */ e.jsx(
    b,
    {
      loading: t,
      error: i,
      loadingComponent: c
    }
  ) : /* @__PURE__ */ e.jsx(
    "div",
    {
      id: "recent-submissions_container",
      ref: a,
      className: "p-2 w-full",
      children: /* @__PURE__ */ e.jsxs("div", { className: "flex flex-col w-full h-full", children: [
        /* @__PURE__ */ e.jsx("div", { style: {
          color: s.secondaryColor
        }, className: "flex items-center justify-between w-full", children: /* @__PURE__ */ e.jsx("h2", { className: "text-lg font-bold", children: "Recent Submissions" }) }),
        /* @__PURE__ */ e.jsx(
          "div",
          {
            id: "recent-submissions_list",
            className: "w-full h-full mt-2",
            children: o.map((r, d) => /* @__PURE__ */ e.jsxs("div", { style: {
              backgroundColor: s.bgColor,
              color: s.secondaryColor
            }, className: "flex items-center justify-between w-full h-10 mt-2 px-2 py-1 rounded-md recent_submission_tile", children: [
              /* @__PURE__ */ e.jsxs("div", { className: "flex items-center", children: [
                /* @__PURE__ */ e.jsx("div", { className: `w-4 h-4 rounded-full ${r.statusDisplay === "Accepted" ? "bg-green-500" : "bg-red-500"} ` }),
                /* @__PURE__ */ e.jsx("p", { style: {
                  color: s.primaryColor
                }, className: "ml-2 text-sm", children: r.title })
              ] }),
              /* @__PURE__ */ e.jsx("p", { className: "text-xs", children: r.statusDisplay })
            ] }, d))
          }
        )
      ] })
    }
  );
}), N = u(j);
export {
  N as default
};
