// react-leetcode/dist/components/userProfileDetails/MiscellaneousDetail.js
import { j as e } from "../../index-CuHmWrv0.js";
import { useState as f, useRef as v, useEffect as b } from "react";
function g({
  icon: h,
  title: s,
  id: r,
  link: a,
  theme: t = {
    primaryColor: "rgba(34,211,238,1)",
    secondaryColor: "rgba(209,213,219,1)"
  }
}) {
  const [l, _] = f(!1), [c, o] = f(!1), n = v(null);
  b(() => {
    var x, p, d;
    const m = (x = n.current) == null ? void 0 : x.offsetWidth, u = (d = (p = n.current) == null ? void 0 : p.parentElement) == null ? void 0 : d.scrollWidth;
    m && u && u <= m && _(!0);
  }, [s]);
  const i = /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
    /* @__PURE__ */ e.jsx("div", { id: r + "icon", className: "misc_content_icon w-5 h-5 flex items-end justify-center", children: h }),
    l && /* @__PURE__ */ e.jsxs("span", { className: "misc_content_title_container group w-full overflow-hidden text-start text-nowrap text-ellipsis", children: [
      /* @__PURE__ */ e.jsx("span", { className: "invisible absolute group-hover:visible group-hover:z-50 rounded text-sm shadow-lg p-2 -ml-[0%] bg-black text-white mt-7", children: s }),
      /* @__PURE__ */ e.jsx("span", { className: "misc_content_title text-base text-start", children: s })
    ] }),
    !l && /* @__PURE__ */ e.jsx("span", { className: "misc_content_title_container overflow-hidden text-start text-nowrap text-ellipsis w-full", children: /* @__PURE__ */ e.jsx("span", { ref: n, className: "misc_content_title text-base text-start", children: s }) })
  ] });
  return a ? /* @__PURE__ */ e.jsx(
    "a",
    {
      style: {
        color: c ? t.primaryColor : t.secondaryColor
      },
      onMouseOver: () => o(!0),
      onMouseOut: () => o(!1),
      id: r,
      className: `misc_content_container flex items-start gap-2 mb-2 hover:text-[${t.primaryColor}]`,
      target: "_blank",
      href: a,
      children: i
    }
  ) : /* @__PURE__ */ e.jsx(
    "div",
    {
      id: r,
      className: "misc_content_container flex items-start gap-2 mb-2",
      style: {
        color: c ? t.primaryColor : t.secondaryColor
      },
      onMouseOver: () => o(!0),
      onMouseOut: () => o(!1),
      children: i
    }
  );
}
export {
  g as default
};
