// react-leetcode/dist/components/userProfileDetails/UserProfileDetails.js
import { j as n } from "../../index-CuHmWrv0.js";
import d, { forwardRef as D, useCallback as H, memo as W } from "react";
import I from "../../utils/leetcodeQuery.js";
import { useFetch as T } from "../../hooks/index.js";
import l from "./MiscellaneousDetail.js";
import { getPath as b } from "../../utils/index.js";
import F from "../LoadingOrError.js";
var C = {
  color: void 0,
  size: void 0,
  className: void 0,
  style: void 0,
  attr: void 0
}, M = d.createContext && /* @__PURE__ */ d.createContext(C), U = ["attr", "size", "title"];
function R(t, e) {
  if (t == null)
    return {};
  var r = G(t, e), i, a;
  if (Object.getOwnPropertySymbols) {
    var o = Object.getOwnPropertySymbols(t);
    for (a = 0; a < o.length; a++)
      i = o[a], !(e.indexOf(i) >= 0) && Object.prototype.propertyIsEnumerable.call(t, i) && (r[i] = t[i]);
  }
  return r;
}
function G(t, e) {
  if (t == null)
    return {};
  var r = {}, i = Object.keys(t), a, o;
  for (o = 0; o < i.length; o++)
    a = i[o], !(e.indexOf(a) >= 0) && (r[a] = t[a]);
  return r;
}
function f() {
  return f = Object.assign ? Object.assign.bind() : function(t) {
    for (var e = 1; e < arguments.length; e++) {
      var r = arguments[e];
      for (var i in r)
        Object.prototype.hasOwnProperty.call(r, i) && (t[i] = r[i]);
    }
    return t;
  }, f.apply(this, arguments);
}
function P(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var i = Object.getOwnPropertySymbols(t);
    e && (i = i.filter(function(a) {
      return Object.getOwnPropertyDescriptor(t, a).enumerable;
    })), r.push.apply(r, i);
  }
  return r;
}
function h(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? P(Object(r), !0).forEach(function(i) {
      K(t, i, r[i]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : P(Object(r)).forEach(function(i) {
      Object.defineProperty(t, i, Object.getOwnPropertyDescriptor(r, i));
    });
  }
  return t;
}
function K(t, e, r) {
  return e = A(e), e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
function A(t) {
  var e = Q(t, "string");
  return typeof e == "symbol" ? e : String(e);
}
function Q(t, e) {
  if (typeof t != "object" || t === null)
    return t;
  var r = t[Symbol.toPrimitive];
  if (r !== void 0) {
    var i = r.call(t, e || "default");
    if (typeof i != "object")
      return i;
    throw new TypeError("@@toPrimitive must return a primitive value.");
  }
  return (e === "string" ? String : Number)(t);
}
function L(t) {
  return t && t.map((e, r) => /* @__PURE__ */ d.createElement(e.tag, h({
    key: r
  }, e.attr), L(e.child)));
}
function c(t) {
  return (e) => /* @__PURE__ */ d.createElement(X, f({
    attr: h({}, t.attr)
  }, e), L(t.child));
}
function X(t) {
  var e = (r) => {
    var {
      attr: i,
      size: a,
      title: o
    } = t, g = R(t, U), u = a || r.size || "1em", s;
    return r.className && (s = r.className), t.className && (s = (s ? s + " " : "") + t.className), /* @__PURE__ */ d.createElement("svg", f({
      stroke: "currentColor",
      fill: "currentColor",
      strokeWidth: "0"
    }, r.attr, i, g, {
      className: s,
      style: h(h({
        color: t.color || r.color
      }, r.style), t.style),
      height: u,
      width: u,
      xmlns: "http://www.w3.org/2000/svg"
    }), o && /* @__PURE__ */ d.createElement("title", null, o), t.children);
  };
  return M !== void 0 ? /* @__PURE__ */ d.createElement(M.Consumer, null, (r) => e(r)) : e(C);
}
function $(t) {
  return c({ tag: "svg", attr: { version: "1.2", baseProfile: "tiny", viewBox: "0 0 24 24" }, child: [{ tag: "path", attr: { d: "M12 5c1.609 0 3.12.614 4.254 1.73 1.126 1.107 1.746 2.579 1.746 4.14s-.62 3.03-1.745 4.139l-4.255 4.184-4.254-4.186c-1.125-1.107-1.745-2.576-1.745-4.139s.62-3.032 1.745-4.141c1.135-1.113 2.647-1.727 4.254-1.727m0-2c-2.047 0-4.096.768-5.657 2.305-3.124 3.074-3.124 8.057 0 11.131l5.657 5.563 5.657-5.565c3.124-3.072 3.124-8.056 0-11.129-1.561-1.537-3.609-2.305-5.657-2.305zM12 8.499c.668 0 1.296.26 1.768.731.976.976.976 2.562 0 3.537-.473.472-1.1.731-1.768.731s-1.295-.26-1.768-.731c-.976-.976-.976-2.562 0-3.537.473-.471 1.101-.731 1.768-.731m0-1c-.896 0-1.792.342-2.475 1.024-1.367 1.367-1.367 3.584 0 4.951.684.684 1.578 1.024 2.475 1.024s1.792-.342 2.475-1.024c1.366-1.367 1.366-3.584 0-4.951-.683-.683-1.579-1.024-2.475-1.024z" }, child: [] }] })(t);
}
function q(t) {
  return c({ tag: "svg", attr: { viewBox: "0 0 24 24" }, child: [{ tag: "path", attr: { fill: "none", d: "M0 0h24v24H0V0z" }, child: [] }, { tag: "path", attr: { d: "M14 6V4h-4v2h4zM4 8v11h16V8H4zm16-2c1.11 0 2 .89 2 2v11c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2l.01-11c0-1.11.88-2 1.99-2h4V4c0-1.11.89-2 2-2h4c1.11 0 2 .89 2 2v2h4z" }, child: [] }] })(t);
}
function J(t) {
  return c({ tag: "svg", attr: { viewBox: "0 0 512 512" }, child: [{ tag: "path", attr: { fill: "none", strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: "32", d: "M32 192 256 64l224 128-224 128L32 192z" }, child: [] }, { tag: "path", attr: { fill: "none", strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: "32", d: "M112 240v128l144 80 144-80V240m80 128V192M256 320v128" }, child: [] }] })(t);
}
function Y(t) {
  return c({ tag: "svg", attr: { viewBox: "0 0 1024 1024" }, child: [{ tag: "path", attr: { d: "M1025.02 512c0-272.016-213.663-495.104-482.319-511.023-5.536-.608-11.088-1.009-16.72-1.009-1.664 0-3.328.176-4.992.224-2.992-.048-5.968-.224-8.992-.224C229.117-.032-1.026 229.664-1.026 512s230.144 512.032 513.023 512.032c3.024 0 6-.176 9.008-.24 1.664.064 3.328.24 4.992.24 5.632 0 11.184-.4 16.72-1.009 268.64-15.92 482.304-238.976 482.303-511.023zm-95.451 164.832c-17.632-5.12-61.92-16.24-140.064-25.392 6.464-44.192 10-90.896 10-139.44 0-38.256-2.208-75.343-6.288-111.008 99.008-11.824 142.384-26.72 145.296-27.745l-11.92-33.584c22.24 53.088 34.56 111.296 34.56 172.336 0 58.193-11.28 113.761-31.583 164.833zM285.488 512.001c0-35.808 2.37-70.77 6.705-104.401 51.888 4.08 113.936 7.088 186.863 7.792v222.064c-70.992.688-131.664 3.568-182.688 7.473-7.04-42.193-10.88-86.88-10.88-132.928zM542.945 68.223c78.464 22.736 145.648 131.695 175.744 276.111-48.368 3.856-106.624 6.673-175.744 7.33V68.223zm-63.886.783V351.63c-68.368-.688-126.88-3.473-176.063-7.232C333.7 201.79 401.428 93.646 479.059 69.006zm0 632.223l.001 253.743c-72.4-22.976-136.192-118.575-169.36-247.023 47.76-3.504 104.096-6.063 169.359-6.72zm63.888 254.543l-.001-254.56c65.952.623 122.064 3.28 169.217 6.928-32.608 130.128-96 226.416-169.216 247.632zm-.001-318.32l.001-222.032c73.311-.688 134.991-3.776 186.191-8a844.922 844.922 0 0 1 6.496 104.592c0 46.128-3.712 90.864-10.528 133.12-50.416-4.08-110.8-7.008-182.16-7.68zm371.858-323.52c-9.664 3.008-50.063 14.48-131.023 24.032-18.048-95.952-50.672-177.968-93.12-237.168C788.197 143.18 867.797 219.1 914.805 313.932zM358.82 90.589c-52.208 59.952-94.832 146.161-118.096 248.113-72.48-7.856-115.921-17.089-133.312-21.281 50.72-104.64 141.04-186.752 251.408-226.832zM83.637 377.182c12.32 3.344 58.913 14.941 145.553 24.525a795.86 795.86 0 0 0-7.68 110.305c0 48.273 4.368 94.721 12.24 138.688-74.4 8.033-120.16 17.649-140.688 22.609-19.44-50.096-30.208-104.447-30.208-161.312 0-46.96 7.312-92.256 20.783-134.815zm37.457 355.166c23.264-4.944 64.912-12.464 126.592-18.928 24.288 89.712 63.792 165.616 111.136 219.968-101.12-36.72-185.296-108.752-237.728-201.04zM690.662 923.18c38.224-53.264 68.48-125.024 87.296-208.801 63.408 7.28 103.216 15.792 123.296 20.864-48.016 83.072-121.855 149.393-210.592 187.937z" }, child: [] }] })(t);
}
function Z(t) {
  return c({ tag: "svg", attr: { viewBox: "0 0 496 512" }, child: [{ tag: "path", attr: { d: "M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z" }, child: [] }] })(t);
}
function t1(t) {
  return c({ tag: "svg", attr: { viewBox: "0 0 448 512" }, child: [{ tag: "path", attr: { d: "M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z" }, child: [] }] })(t);
}
function e1(t) {
  return c({ tag: "svg", attr: { viewBox: "0 0 512 512" }, child: [{ tag: "path", attr: { d: "M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z" }, child: [] }] })(t);
}
function r1(t) {
  return c({ tag: "svg", attr: { viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round" }, child: [{ tag: "path", attr: { d: "M20.59 13.41l-7.17 7.17a2 2 0 0 1-2.83 0L2 12V2h10l8.59 8.59a2 2 0 0 1 0 2.82z" }, child: [] }, { tag: "line", attr: { x1: "7", y1: "7", x2: "7.01", y2: "7" }, child: [] }] })(t);
}
const i1 = D(({
  userName: t,
  theme: e = {
    primaryColor: "rgba(34,211,238,1)",
    secondaryColor: "rgba(209,213,219,1)",
    bgColor: "rgba(68,64,60,1)"
  },
  loadingComponent: r,
  showRank: i = !0,
  hideLocation: a = !1,
  hideSchool: o = !1,
  hideWebsite: g = !1,
  hideCompany: u = !1,
  hideSkills: s = !1
}, N) => {
  const V = H(() => I.fetchUserProfile(t), [t]), { data: m, loading: y, error: z } = T(V);
  if (y || z || !m)
    return /* @__PURE__ */ n.jsx(
      F,
      {
        loading: y,
        error: z,
        loadingComponent: r
      }
    );
  const { userAvatar: E, realName: S, ranking: B, countryName: w, company: O, school: _, skillTags: k, websites: v } = m.profile, { githubUrl: p, linkedinUrl: x, twitterUrl: j } = m;
  return /* @__PURE__ */ n.jsxs("div", { style: {
    background: e.bgColor,
    color: e.secondaryColor
  }, id: "user_profile_detail_container", ref: N, className: "w-full shadow-md flex flex-col rounded-2xl items-start p-4", children: [
    /* @__PURE__ */ n.jsxs("div", { id: "user_main_content", className: "flex justify-center gap-4", children: [
      /* @__PURE__ */ n.jsx(
        "img",
        {
          id: "user_avatar",
          src: E,
          alt: `${t}'s avatar`,
          className: "w-20 h-20 rounded-md"
        }
      ),
      /* @__PURE__ */ n.jsxs(
        "div",
        {
          id: "user_name_and_rank_container",
          className: "flex flex-col items-start",
          children: [
            /* @__PURE__ */ n.jsx("p", { style: {
              color: e.primaryColor
            }, id: "user_real_name", className: "text-xl font-semibold", children: S }),
            /* @__PURE__ */ n.jsx("p", { id: "user_name", className: "text-sm", children: t }),
            i && /* @__PURE__ */ n.jsxs("div", { id: "rank_details", className: "flex gap-2 text-sm mt-auto", children: [
              /* @__PURE__ */ n.jsx("p", { style: {
                color: e.secondaryColor
              }, id: "rank_label", children: "Rank :" }),
              /* @__PURE__ */ n.jsx("p", { id: "rank_value", className: "font-semibold", children: B })
            ] })
          ]
        }
      )
    ] }),
    /* @__PURE__ */ n.jsx("p", { className: "text-sm font-semibold my-3 text-start text-gray-300", children: m.profile.aboutMe }),
    /* @__PURE__ */ n.jsxs(
      "div",
      {
        id: "user_misc_details",
        className: "overflow-hidden w-full text-ellipsis",
        children: [
          O && !u && /* @__PURE__ */ n.jsx(
            l,
            {
              id: "user_company",
              icon: /* @__PURE__ */ n.jsx(q, {}),
              title: O
            }
          ),
          w && !a && /* @__PURE__ */ n.jsx(
            l,
            {
              id: "user_location",
              icon: /* @__PURE__ */ n.jsx($, {}),
              title: w
            }
          ),
          _ && !o && /* @__PURE__ */ n.jsx(
            l,
            {
              id: "user_school",
              icon: /* @__PURE__ */ n.jsx(J, {}),
              title: _
            }
          ),
          v && !g && /* @__PURE__ */ n.jsx(
            l,
            {
              id: "user_website",
              icon: /* @__PURE__ */ n.jsx(Y, {}),
              title: v[0],
              link: v[0]
            }
          ),
          p && /* @__PURE__ */ n.jsx(
            l,
            {
              id: "user_github",
              icon: /* @__PURE__ */ n.jsx(Z, {}),
              title: b(p),
              link: p
            }
          ),
          x && /* @__PURE__ */ n.jsx(
            l,
            {
              id: "user_linkedin",
              icon: /* @__PURE__ */ n.jsx(t1, {}),
              title: b(x),
              link: x
            }
          ),
          j && /* @__PURE__ */ n.jsx(
            l,
            {
              id: "user_twitter",
              icon: /* @__PURE__ */ n.jsx(e1, {}),
              title: b(j),
              link: j
            }
          ),
          k && !s && /* @__PURE__ */ n.jsx(
            l,
            {
              id: "user_twitter",
              icon: /* @__PURE__ */ n.jsx(r1, { className: "-rotate-90" }),
              title: k.join(", ")
            }
          )
        ]
      }
    )
  ] });
}), u1 = W(i1);
export {
  u1 as default
};
