// react-leetcode/dist/components/heatMap/HeatMap.js
import { j as e } from "../../index-CuHmWrv0.js";
import { forwardRef as g, useState as x, useRef as j, useCallback as b, useEffect as w, memo as y } from "react";
import C from "../../utils/leetcodeQuery.js";
import { useFetch as _ } from "../../hooks/index.js";
import k from "../LoadingOrError.js";
import { calcOpacity as v } from "../../utils/index.js";
const M = ({
  count: r,
  width: s
}) => /* @__PURE__ */ e.jsx(e.Fragment, { children: /* @__PURE__ */ e.jsx(
  "div",
  {
    style: {
      backgroundColor: `rgba(20, 255, 20, ${v(r)})`,
      width: `${s}px`
    },
    className: "sq-block aspect-square rounded-sm border border-black border-opacity-30"
  }
) }), R = g(({
  userName: r,
  loadingComponent: s,
  theme: l = {
    primaryColor: "rgba(34,211,238,1)",
    secondaryColor: "rgba(209,213,219,1)",
    bgColor: "rgba(68,64,60,1)"
  },
  showUserName: c = !1
}, f) => {
  const [m, p] = x(69), n = j(null), u = b(() => C.fetchUserHeatMap(r), [r]), { data: t, loading: i, error: d } = _(u);
  if (w(() => {
    var a;
    const o = (a = n.current) == null ? void 0 : a.getBoundingClientRect().width;
    p(o || 69);
  }, [i]), i || d || !t)
    return /* @__PURE__ */ e.jsx(
      k,
      {
        loading: i,
        error: d,
        loadingComponent: s
      }
    );
  const h = t.length - 1;
  return /* @__PURE__ */ e.jsx("div", { ref: n, className: "w-full", children: /* @__PURE__ */ e.jsxs(
    "div",
    {
      id: "heat_map_container",
      style: {
        background: l.bgColor
      },
      ref: f,
      className: "w-full gap-1 rounded-lg p-4 flex flex-col items-center justify-center",
      children: [
        /* @__PURE__ */ e.jsxs(
          "h2",
          {
            id: "heat_map_title",
            style: { color: l.primaryColor },
            className: "flex justify-between font-medium w-full",
            children: [
              /* @__PURE__ */ e.jsx("span", { children: "Heatmap (Last 52 Weeks)" }),
              c && /* @__PURE__ */ e.jsx("span", { children: r })
            ]
          }
        ),
        /* @__PURE__ */ e.jsx(
          "div",
          {
            id: "heat_map",
            className: "w-full h-full grid grid-rows-7 grid-flow-col justify-center",
            style: {
              rowGap: "1px",
              columnGap: "2px"
            },
            children: t.map((o, a) => /* @__PURE__ */ e.jsx(
              M,
              {
                width: m / 69,
                i: a,
                count: o.submissionCount,
                date: o.date
              },
              a
            ))
          }
        ),
        /* @__PURE__ */ e.jsxs(
          "div",
          {
            id: "heat_map_date_range",
            style: { color: l.secondaryColor },
            className: "flex justify-between w-full",
            children: [
              /* @__PURE__ */ e.jsx("span", { children: t[0].date }),
              /* @__PURE__ */ e.jsx("span", { children: t[h - 1].date })
            ]
          }
        )
      ]
    }
  ) });
}), B = y(R);
export {
  B as default
};
