// react-leetcode/dist/components/contestInfo/ContestStaticData.js
import { j as o } from "../../index-CuHmWrv0.js";
function n({
  contestData: r,
  theme: l = {
    primaryColor: "rgba(34,211,238,1)",
    secondaryColor: "rgba(209,213,219,1)",
    bgColor: "rgba(68,64,60,1)"
  }
}) {
  return /* @__PURE__ */ o.jsxs("div", { id: "overall_contest_info_container", className: "flex flex-wrap mb-1 gap-1 justify-between p-2", children: [
    /* @__PURE__ */ o.jsxs("div", { id: "contest_overall_rating", children: [
      /* @__PURE__ */ o.jsx("h2", { style: { color: l.secondaryColor }, className: "font-semibold", children: "Contest Rating" }),
      /* @__PURE__ */ o.jsx("span", { style: { color: l.primaryColor }, className: "text-3xl", children: Math.round(r.rating) })
    ] }),
    /* @__PURE__ */ o.jsxs("div", { id: "contest_global_ranking", children: [
      /* @__PURE__ */ o.jsx("p", { style: { color: l.secondaryColor }, children: "Global Ranking" }),
      /* @__PURE__ */ o.jsx("span", { style: { color: l.primaryColor }, className: "font-semibold", children: r.globalRanking.toLocaleString() }),
      /* @__PURE__ */ o.jsx("span", { style: { color: l.secondaryColor }, className: "font-semibold", children: " /" + r.totalParticipants.toLocaleString() })
    ] }),
    /* @__PURE__ */ o.jsxs("div", { id: "total_attended_contest", children: [
      /* @__PURE__ */ o.jsx("p", { style: { color: l.secondaryColor }, children: "Attended" }),
      /* @__PURE__ */ o.jsx("span", { style: { color: l.primaryColor }, children: r.attendedContestsCount })
    ] })
  ] });
}
export {
  n as default
};
