// react-leetcode/dist/components/contestInfo/DynamicContestData.js
import { j as o } from "../../index-CuHmWrv0.js";
function e({
  contestData: s,
  theme: r = {
    primaryColor: "rgba(34,211,238,1)",
    secondaryColor: "rgba(209,213,219,1)",
    bgColor: "rgba(68,64,60,1)"
  }
}) {
  const l = new Date(s.contest.startTime * 1e3);
  return /* @__PURE__ */ o.jsxs("div", { id: "particular_contest_detail_container", className: "flex flex-wrap mb-1 gap-1 justify-between p-2", children: [
    /* @__PURE__ */ o.jsxs("div", { id: "contest_rank", children: [
      /* @__PURE__ */ o.jsx("h2", { style: { color: r.secondaryColor }, className: "font-semibold", children: "Contest Ranking" }),
      /* @__PURE__ */ o.jsx("span", { style: { color: r.primaryColor }, className: "text-3xl", children: Math.round(s.rating) })
    ] }),
    /* @__PURE__ */ o.jsxs("div", { id: "contest_date", children: [
      /* @__PURE__ */ o.jsx("p", { style: { color: r.secondaryColor }, children: l.toLocaleDateString() }),
      /* @__PURE__ */ o.jsx("span", { style: { color: r.primaryColor }, className: "font-semibold", children: s.contest.title })
    ] }),
    /* @__PURE__ */ o.jsxs("div", { id: "contest_rank", children: [
      /* @__PURE__ */ o.jsx("p", { style: { color: r.secondaryColor }, children: "Rank" }),
      /* @__PURE__ */ o.jsx("span", { style: { color: r.primaryColor }, children: s.ranking })
    ] }),
    /* @__PURE__ */ o.jsxs("div", { id: "total_solved_questions", children: [
      /* @__PURE__ */ o.jsx("p", { style: { color: r.secondaryColor }, children: "Solved" }),
      /* @__PURE__ */ o.jsx("span", { style: { color: r.primaryColor }, children: s.problemsSolved }),
      /* @__PURE__ */ o.jsx("span", { style: { color: r.secondaryColor }, children: " /4" })
    ] })
  ] });
}
export {
  e as default
};
