// react-leetcode/dist/components/contestInfo/UserContestInfo.js
import { j as t } from "../../index-CuHmWrv0.js";
import { forwardRef as g, useState as C, useCallback as p, useMemo as x, memo as y } from "react";
import { useFetch as b } from "../../hooks/index.js";
import h from "../../utils/leetcodeQuery.js";
import j from "./ContestGraph.js";
import { getCordinates as D } from "../../utils/index.js";
import k from "./ContestStaticData.js";
import R from "../LoadingOrError.js";
import I from "./DynamicContestData.js";
const v = g(({
  userName: s,
  theme: i = {
    primaryColor: "rgba(34,211,238,1)",
    secondaryColor: "rgba(209,213,219,1)",
    bgColor: "rgba(68,64,60,1)"
  },
  loadingComponent: m
}, c) => {
  const [o, d] = C(-1), f = p(() => h.fetchUserContestDetails(s), [s]), { data: r, loading: n, error: a } = b(f), { userContestRankingHistory: e, userContestRanking: l } = r || {}, u = x(() => e ? D(e) : [], [e]);
  return !r || n || a ? /* @__PURE__ */ t.jsx(R, { error: a, loading: n, loadingComponent: m }) : /* @__PURE__ */ t.jsxs(
    "div",
    {
      id: "user_contest_info_container",
      ref: c,
      className: "w-full p-4 rounded-lg h-full",
      style: { background: i.bgColor },
      children: [
        o === -1 ? /* @__PURE__ */ t.jsx(k, { contestData: l }) : /* @__PURE__ */ t.jsx(I, { contestData: r.userContestRankingHistory[o] }),
        /* @__PURE__ */ t.jsx(
          j,
          {
            activeIndex: o,
            setActiveIndex: d,
            data: u
          }
        )
      ]
    }
  );
}), A = y(v);
export {
  A as default
};
