// react-leetcode/dist/utils/queries.js
const e = `
query ($username: String!) {
    matchedUser(username: $username) {
        username
        socialAccounts
        githubUrl
        linkedinUrl
        twitterUrl
        profile {
            realName
            websites
            countryName
            skillTags
            company
            school
            starRating
            aboutMe
            userAvatar
            reputation
            ranking
        }
        activeBadge {
            id
            displayName
            icon
        }
    }
}
`, t = `
query ($username: String!) {
    userContestRanking(username: $username) {
        attendedContestsCount
        rating
        globalRanking
        totalParticipants
        topPercentage
        badge {
            name
        }
    }
    userContestRankingHistory(username: $username) {
        attended
        trendDirection
        problemsSolved
        totalProblems
        finishTimeInSeconds
        rating
        ranking
        contest {
            title
            startTime
        }
    }
}
`, s = `
query ($username: String!) {
    allQuestionsCount {
        difficulty
        count
    }
    matchedUser(username: $username) {
        profile {
            ranking
        }
        submitStats {
            acSubmissionNum {
                difficulty
                count
                submissions
            }
            totalSubmissionNum {
                difficulty
                count
                submissions
            }
        }
    }
}
`, n = `
query ($username: String!) {
    recentSubmissionList(username: $username, limit: 10) {
        title
        titleSlug
        timestamp
        statusDisplay
        lang
    }
}
`, a = `
query ($username: String!) {
    matchedUser(username: $username) {
        submissionCalendar
    }
}
`;
export {
  t as contestDetailsQuery,
  a as heatMapQuery,
  e as profileQuery,
  n as recentSubmissionQuery,
  s as solvedProblemsStatsQuery
};
