// react-leetcode/dist/components/solvedProblemsStat/OverallProgress.js
import { j as e } from "../../index-CuHmWrv0.js";
import { useState as f } from "react";
function h({
  totalQuestions: c,
  totalSolved: s,
  primaryColor: l = "rgba(34,211,238,1)",
  secondaryColor: p = "rgba(209,213,219,1)"
}) {
  const [o, a] = f(!1), t = (s / c * 100).toFixed(2), r = 55, i = 2 * Math.PI * r, n = i * ((100 - parseFloat(t)) / 100), x = t.split(".")[0], d = t.split(".")[1] + "%";
  return /* @__PURE__ */ e.jsxs(
    "svg",
    {
      id: "circular_progress_container",
      onMouseOver: () => {
        a(!0);
      },
      onMouseOut: () => {
        a(!1);
      },
      viewBox: "0 0 160 160",
      style: { transform: "rotate(-90deg)" },
      className: "h-[75%] w-[75%] flex items-start justify-start",
      children: [
        /* @__PURE__ */ e.jsx(
          "circle",
          {
            id: "base_circle",
            r,
            cx: "80",
            cy: "80",
            fill: "transparent",
            stroke: "#e0e0e0",
            strokeWidth: "5px",
            opacity: 0.3
          }
        ),
        /* @__PURE__ */ e.jsx(
          "circle",
          {
            id: "progress_circle",
            r,
            cx: "80",
            cy: "80",
            fill: "transparent",
            stroke: "rgb(202,138,4)",
            strokeWidth: "6px",
            strokeDasharray: i,
            strokeDashoffset: n,
            strokeLinecap: "round",
            style: { borderRadius: "50%" },
            children: /* @__PURE__ */ e.jsx("animate", { id: "animate_progress_circle", attributeName: "stroke-dashoffset", values: `360;${n}`, dur: "2s" })
          }
        ),
        /* @__PURE__ */ e.jsxs(
          "text",
          {
            id: "progress_percentage",
            x: "50%",
            y: "50%",
            fill: l,
            fontSize: "30px",
            textAnchor: "middle",
            fontWeight: "bold",
            style: { transform: "rotate(90deg) translate(0, -156px)" },
            children: [
              o ? x : s,
              o && /* @__PURE__ */ e.jsx("tspan", { fontSize: "12px", children: "." + d })
            ]
          }
        ),
        /* @__PURE__ */ e.jsx(
          "text",
          {
            id: "progress_label",
            x: "62px",
            y: "100px",
            fill: p,
            fontSize: "12px",
            style: { transform: "rotate(90deg) translate(0, -156px)" },
            children: "solved"
          }
        )
      ]
    }
  );
}
export {
  h as default
};
