// react-leetcode/dist/components/solvedProblemsStat/SolvedProblemsStats.js
import { j as s } from "../../index-CuHmWrv0.js";
import { forwardRef as v, useCallback as _, memo as C } from "react";
import { useFetch as N } from "../../hooks/index.js";
import w from "../../utils/leetcodeQuery.js";
import S from "../LoadingOrError.js";
import h from "./OverallProgress.js";
const k = v(({
  userName: r,
  loadingComponent: d,
  theme: o = {
    primaryColor: "rgba(34,211,238,1)",
    secondaryColor: "rgba(209,213,219,1)",
    bgColor: "rgba(68,64,60,1)"
  },
  showUserName: t = !0
}, m) => {
  const u = _(() => w.fetchUserSolvedProblemsStats(r), [r]), { data: e, loading: a, error: n } = N(u);
  if (a || n || !e)
    return /* @__PURE__ */ s.jsx(
      S,
      {
        loading: a,
        error: n,
        loadingComponent: d
      }
    );
  const [f, ...p] = (e == null ? void 0 : e.allQuestionsCount) || [], [b, ...x] = (e == null ? void 0 : e.acSubmissionNum) || [], i = {
    Easy: "bg-green-600",
    Medium: "bg-amber-300",
    Hard: "bg-red-500"
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  };
  return /* @__PURE__ */ s.jsxs(
    "div",
    {
      id: "solved_problems_stats_container",
      ref: m,
      className: "w-full flex flex-col items-center rounded-xl px-4 py-2 gap-2",
      style: { background: o.bgColor },
      children: [
        /* @__PURE__ */ s.jsxs("div", { id: "solved_problems_stats_label", className: "w-full flex justify-between p-3", children: [
          /* @__PURE__ */ s.jsx("span", { className: "text-sm font-semibold", style: { color: o.secondaryColor }, children: t ? r : "Solved Problems" }),
          t && /* @__PURE__ */ s.jsx("span", { className: "text-sm font-semibold pr-1", style: { color: o.secondaryColor }, children: "#" + e.rank })
        ] }),
        /* @__PURE__ */ s.jsxs("div", { id: "solved_problems_stats_progress_deails", className: "w-full flex justify-between items-center", children: [
          /* @__PURE__ */ s.jsx(
            h,
            {
              totalQuestions: f.count,
              totalSolved: b.count,
              primaryColor: "#fff"
            }
          ),
          /* @__PURE__ */ s.jsx("div", { id: "linear_progress_container", className: "w-full", children: p.map((l, g) => {
            const c = x[g], y = l.count, j = c.count / y * 100;
            return /* @__PURE__ */ s.jsxs("div", { id: `progress_bar_${l.difficulty}`, className: "progress_bar mt-3 first:mt-0 w-full", children: [
              /* @__PURE__ */ s.jsxs("div", { className: "flex justify-between px-1", children: [
                /* @__PURE__ */ s.jsx("span", { className: "text-sm", style: { color: o == null ? void 0 : o.secondaryColor }, children: l.difficulty }),
                /* @__PURE__ */ s.jsxs("span", { className: "w-[4.5rem] text-end", children: [
                  /* @__PURE__ */ s.jsx("span", { className: "font-semibold", style: { color: o.primaryColor }, children: c.count }),
                  /* @__PURE__ */ s.jsx("span", { className: "text-xs pb-2", style: { color: o.secondaryColor }, children: " /" + l.count })
                ] })
              ] }),
              /* @__PURE__ */ s.jsx("div", { className: `${i[l.difficulty]} progress_label bg-opacity-20 w-full  rounded-full h-2 dark:bg-gray-700`, children: /* @__PURE__ */ s.jsx("div", { style: { width: `${j}%` }, children: /* @__PURE__ */ s.jsx("div", { className: `${i[l.difficulty]} animate-slide h-2 rounded-full dark:bg-blue-500` }) }) })
            ] }, l.difficulty);
          }) })
        ] })
      ]
    }
  );
}), O = C(k);
export {
  O as default
};
